File: server/packages/graphql/src/model/comment.ts
import mongoose from "mongoose";
const Schema = mongoose.Schema;

const comment = new Schema({
  userId: {
    type: String,
    required: "user id required"
  },
  postId: {
    type: String,
    required: "article id required"
  },
  authorName: {
    type: String,
    required: "author name is required"
  },
  text: {
    type: String,
    required: "text is required"
  },
  date: {
    type: Date,
    default: Date.now,
  },
});

export default mongoose.model("comments", comment);

File: client/lib/domain/entities/comment/comment.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:flutter/foundation.dart';

part 'comment.freezed.dart';
part 'comment.g.dart';

@freezed
class Comment with _$Comment{
  const factory Comment({
    required int id,
    required int userId,
    required int postId,
    required String authorName,
    required String text,
    required DateTime date,
  })  = _Comment;

  factory Comment.fromJson(Map<String, Object?> json) => _$CommentFromJson(json);
}
